### DISCUSSION TITLE
Diarization not working well

### DISCUSSION THREAD

**Discussion by marscrusader** [deepgram/community#149] [General help]
Hi I just tested the both the beta(api.beta.deepgram.com) and production endpoint with multiple different audio files:
```
curl --request POST \
  --url 'https://api.deepgram.com/v1/listen?model=whisper-large&diarize=true&punctuate=true' \
  --header 'Authorization: Token {TOKEN}' \
  --header 'Content-Type: application/json' \
  --data '{"url": "https://voiceage.com/wbsamples/in_mono/Conference.wav"}'
```
I used both nova and whisper models and both were unfortunately disappointing in terms of speaker recognition. The endpoint is unable to identify different speakers and keeps returning `"speaker": 0` for all audio files i tested with. Additionally, i find that the nova model is not as accurate in terms of transcription compared to whisper in most cases but this is not important as i can always use `whisper-large`.

Does anyone have any experience using `diarize=true`? Did it correctly identify different speakers? 

**Update**: I also found that when using whisper, the speaker diarization does not work well for certain audio files compared to nova. Example: 
```
curl --request POST \
  --url 'https://api.deepgram.com/v1/listen?diarize=true&punctuate=true&model=whisper-large' \
  --header 'Authorization: Token {TOKEN}' \
  --header 'Content-Type: application/json' \
  --data '{"url": "https://res.cloudinary.com/deepgram/video/upload/v1663090404/dg-audio/AI_Show_afpqeh.m4a"}'
```
Try the above curl request using `model=nova` and `model=whisper-large` to see the difference. When using whisper, the endpoint returns `"speaker": 0` for all speakers. When using nova, the endpoint is able to differentiate between the speakers. The audio file is taken from deepgram.com.

**Comment by incubatize** [comment]
Hi @marscrusader  - appreciate the feedback. Do you see the same issue across all the files? Or just the one you've highlighted in the code? Let me also review this internally with our team. 
thanks again for the detailed feedback.

**Reply by marscrusader** [reply]
Hi @incubatize, thanks for replying. I see the issue happening across some files but not all. I have only tested the endpoint with audio files recorded using my macbook's speaker and files that are available online such as [this](https://replicate.delivery/pbxt/IZjTvet2ZGiyiYaMEEPrzn0xY1UDNsh0NfcO9qeTlpwCo7ig/lex-levin-4min.mp3) and [this](https://res.cloudinary.com/deepgram/video/upload/v1663090404/dg-audio/AI_Show_afpqeh.m4a). Perhaps this might be obvious but i noticed that files recorded with my macbook's mic are not as accurate as files recorded by professional mics.

What i also found is that nova's transcription is not as "clean" as whisper's(removing filler words, removing words that are repeated twice, correcting grammar error) and not as accurate. Here's an example:
```
curl --request POST \
  --url 'https://api.deepgram.com/v1/listen?diarize=true&punctuate=true&model=whisper-large' \
  --header 'Authorization: Token {TOKEN}' \
  --header 'Content-Type: application/json' \
  --data '{"url": "https://replicate.delivery/pbxt/IZjTvet2ZGiyiYaMEEPrzn0xY1UDNsh0NfcO9qeTlpwCo7ig/lex-levin-4min.mp3"}'
```
In the above audio file, the podcast host Lex Friedman says "what do you hope to build?" in the first 30 seconds. Whisper correctly transcribed this but Nova transcribed this to "You hope to build?".

Overall conclusion:
Nova - Fast, okay-ish speaker diarization, transcription not as good as whisper-large.
Whisper-large - Slow, not very good speaker diarization, great transcription.

**Comment by wuzimi** [comment]
Same issue here. Where is customer support from Deepgram?

**Comment by allprogrammers** [comment]
I am facing the same issue. 

The diarization isn't working. Neither for prerecorded nor live streaming.

**Comment by remi-quinto** [comment]
Hi, thanks for your work Deepgram team. 
I see a similar issue here using an mp3 file with two speakers (one single channel), recorded from two different microphones. 
A human can clearly make the difference between the 2 speakers but not the Deepgram service. 
Tested with the API, node.js library and playground. Tested several models, tiers, and versions. 
I can provide the mp3 if that helps (500 KB).

**Reply by incubatize** [reply]
Hi @remi-quinto  - yes, please. It'll be helpful to have the audio file. Assuming you are using batch mode, can you confirm?

**Reply by remi-quinto** [reply]
Thanks for your answer. Here goes the mp3. We are not using batch mode (I can also provide the code, except API-key). 
[test-martin.mp3.zip](https://github.com/deepgram/community/files/11983132/test-martin.mp3.zip)

**Comment by shirgoldbird** [comment]
Hi all, I wanted to let you know that we have a new live-streaming diarization model currently in development. If you're using Deepgram for live-streaming audio with diarization and would like to test the new model out, please send me an email: shir(dot)goldberg(at)deepgram(dot)com.

This is _not_ for pre-recorded audio—only real-time audio.

**Comment by bobandrew** [comment]
Hi deepgram team,
is there any news about diarization improvements? We have the same problem and it always returns Speaker=0. 
For example, we tested with these filses:
https://adv132664224324665576.blob.core.windows.net/temporary/Conversation.wav
https://adv132664224324665576.blob.core.windows.net/temporary/Conversation2.wav
we tried to use Nova, Nova-2 and base (general).

**Comment by natalierutgers** [comment]
Hi @bobandrew. I noticed these conversations are TTS-generated. Our speaker diarization has been trained on human voices, so I suspect that's throwing off the clustering mechanism.

**Reply by bobandrew** [reply]
@natalierutgers , great thanks, it really works with human voice, fine!
